libcudacxx/test/libcudacxx/std/containers/sequences/inplace_vector/assignment.pass.cpp

```diff
@@ -0,0 +1,253 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++11
+
+#include <cuda/std/__algorithm_>
+#include <cuda/std/cassert>
+#include <cuda/std/initializer_list>
+#include <cuda/std/inplace_vector>
+#include <cuda/std/type_traits>
+
+#include "test_iterators.h"
+#include "test_macros.h"
+#include "types.h"
+
+#ifndef TEST_HAS_NO_EXCEPTIONS
+#  include <stdexcept>
+#endif // !TEST_HAS_NO_EXCEPTIONS
+
+template <class T>
+__host__ __device__ constexpr void test_copy()
+{
+  // Zero capacity inplace_vector is nothrow_copy_assignable
+  static_assert(cuda::std::is_nothrow_copy_assignable<cuda::std::inplace_vector<T, 0>>::value, "");
+  static_assert(cuda::std::is_nothrow_copy_assignable<cuda::std::inplace_vector<T, 42>>::value
+                  == cuda::std::conjunction<cuda::std::is_nothrow_copy_constructible<T>,
+                                            cuda::std::is_nothrow_copy_assignable<T>>::value,
+                "");
+
+  { // inplace_vector<T, 0> can be copy assigned
+    const cuda::std::inplace_vector<T, 0> input{};
+    cuda::std::inplace_vector<T, 0> no_capacity{};
+    no_capacity = input;
+    assert(no_capacity.empty());
+  }
+
+  using inplace_vector = cuda::std::inplace_vector<T, 42>;
+  { // inplace_vector<T, N> can be copy assigned an empty input
+    const inplace_vector input{};
+    inplace_vector vec{};
+    vec = input;
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N> can be copy assigned an empty input, shrinking
+    const inplace_vector input{};
+    inplace_vector vec{T(1), T(42), T(1337), T(0)};
+    vec = input;
+    assert(vec.empty());
+  }
+
+  { // inplace_vector<T, N> can be copy assigned a non-empty input, growing from empty
+    const inplace_vector input{T(1), T(42), T(1337), T(0)};
+    inplace_vector vec{};
+    vec = input;
+    assert(!vec.empty());
+    assert(equal_range(vec, input));
+  }
+
+  { // inplace_vector<T, N> can be copy assigned a non-empty input, shrinking
+    const inplace_vector input{T(1), T(42), T(1337), T(0)};
+    inplace_vector vec{T(0), T(42), T(1337), T(42), T(5)};
+    vec = input;
+    assert(!vec.empty());
+    assert(equal_range(vec, input));
+  }
+
+  { // inplace_vector<T, N> can be copy assigned a non-empty input, growing
+    const inplace_vector input{T(1), T(42), T(1337), T(0)};
+    inplace_vector vec{T(0), T(42)};
+    vec = input;
+    assert(!vec.empty());
+    assert(equal_range(vec, input));
+  }
+}
+
+template <class T>
+__host__ __device__ constexpr void test_move()
```

Comment:
I am somewhat missing a test that moving an inplace_vector also moves the elements. E.g., moving an `inplace_vector<std::vector<int>>`, in case it is not empty afterward, will contain empty vectors afterwards. Similarly we could try moving an `inplace_vector<std::unique_ptr<int>>`.

Reply:
Added a test with a move-only type